lib/dir_model/import/path.rb

```diff
@@ -0,0 +1,80 @@
+module DirModel
+  module Import
+    class Path # Csv
+
+      attr_reader :path
+      attr_reader :index
+      attr_reader :current_path
+      attr_reader :previous_path
+
+      def initialize(path)
+        @path = path
+        reset!
+      end
+
+      def size
+        @size ||= ruby_path.size
+      end
+
+      def reset!
+        @index = -1
+        @current_path = @ruby_path = @previous_path = nil
+      end
+
+      def start?
+        index == -1
+      end
+
+      def end?
+        index.nil?
+      end
+
+      def next_path
+        @next_path ||= _read_path
+      end
+
+      def read_path
+        @previous_path = @current_path
+        unless caching_value?
+          @current_path = _read_path { forward! }
+        end
+        current_path
+      end
```

Comment:
i wish my `csv_row_model`'s `csv` code was simpler, but it had constraints that pushed it to be complicated. in this case, i needed this to say "if the next row is not a child, stop parsing children". you have no need to do a "look ahead" operation.